stingray/covariancespectrum.py

```diff
@@ -0,0 +1,467 @@
+            computed with their individual reference band. The covaraince
+            values are normalized.
+
+        unnorm_covar : np.ndarray
+            An array of arrays with mid point band_interest and their
+            covariance. It is the array-form of the dictionary `energy_covar`.
+            The covariance values are unnormalized.
+
+        covar : np.ndarray
+            Normalized covaraiance spectrum.
+
+        covar_error : np.ndarray
+            Errors of the normalized covariance spectrum.
+
+        min_time : int
+            Time of arrival of the earliest photon.
+
+        max_time : int
+            Time of arrival of the last photon.
+
+        min_energy : float
+            Energy of the photon with the minimum energy.
+
+        max_energy : float
+            Energy of the photon with the maximum energy.
+
+        """
+        # Set parameter to distinguish between parent class and derived class.
+        self.avg_covar = True
+
+        self._init_vars(event_list, dt, band_interest, ref_band_interest, std)
+        self.segment_size = segment_size
+
+        self._make_averaged_covar_spectrum()
+
+    def _make_averaged_covar_spectrum(self):
+        """
+        Calls methods from base class for every segment and calculates averaged
+        covariance and error.
+        """
+        self.nbins = int((self.max_time - self.min_time + 1) / self.segment_size)
+
+        for n in range(self.nbins):
+            tstart = self.min_time + n*self.segment_size
+            tend = self.min_time + self.segment_size*(n+1) - 1
+            indices = np.intersect1d(np.where(self.event_list_T[0] >= tstart),
+                                     np.where(self.event_list_T[0] <= tend))
+
+            # Set minimum and maximum values for the specified indices value
+            self._init_special_vars(T_start=indices[0], T_end=indices[-1]+1)
+
+            energy_events = {}
+
+            self._construct_energy_events(energy_events, T_start=indices[0],
+                                          T_end=indices[-1]+1)
+
+            self._update_energy_events(energy_events)
+
+            energy_covar = {}
+            xs_var = {}
+            covar_error = {}
+            self._construct_energy_covar(energy_events, energy_covar,
+                                         covar_error, xs_var)
+
+            if n == 0:  # Declare
+                self.energy_covar = energy_covar
+                self.xs_var = xs_var
+                self.covar_error = covar_error
+            else:  # Sum up
+                for key in energy_covar.keys():
+                    self.energy_covar[key] = self.energy_covar.get(key, 0) + \
+                                             energy_covar[key]
+                    self.xs_var[key] = self.xs_var.get(key, 0) + xs_var[key]
+                    self.covar_error[key] = self.covar_error.get(key, 0) + \
+                                            covar_error[key]
```

Comment:
I'm still worried about this implementation of the error for the covariance spectrum. 

If I understand this code correctly, you compute here the error on covariance spectrum for *every* segment, and then average the errors across the segments. While this is the correct approach for the covariance spectrum *itself*, this is not the right approach for the error on the covariance spectrum.

Following equation 15 of [this paper](http://arxiv.org/pdf/1405.6575v2.pdf), I think we should be calculating the averaged covariance in the way that it's done above, but then use this averaged covariance to compute the error on the averaged covariance as in equation 15. I think for the other quantities in that equation, the full light curve should be used (instead of segments), and `KM` should be the number of segments used.

Does this make sense? @abigailStev @OrkoHunter